core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BillingProfileManagerDAO.scala

```diff
@@ -104,18 +109,18 @@ class BillingProfileManagerDAOImpl(samDAO: SamDAO,
       case Some(value) => value
     }
 
-    for {
-      billingProfiles <- getAllBillingProfiles(azureConfig, ctx)
-    } yield billingProfiles.filter { bp =>
-      samUserResources.map(_.resourceId).contains(bp.projectName.value)
+    @tailrec
```

Comment:
This will be optimized by the compiler regardless of the annotation -  it just makes it so we'll get a compiler warning if it's not a proper tail call.